src/ch02/trafficLight.test.ts

```diff
@@ -10,28 +14,49 @@ class Car {
   }
 }
 
-const CYCLE = [TrafficLight.RED, TrafficLight.YELLOW, TrafficLight.GREEN];
+interface TrafficLight {
+  isRed(): boolean;
+  isYellow(): boolean;
+  isGreen(): boolean;
+  updateCar(): boolean;
+}
 
-const updateCarForLight = (current: TrafficLight) => {
-  const car = new Car();
+class Red implements TrafficLight {
+  isRed() { return true; }
+  isYellow() { return false; }
+  isGreen() { return false; }
+  updateCar() { return car.stop(); }
+}
 
-  if (current === TrafficLight.RED) {
-    return car.stop();
-  }
-  return car.drive();
-};
+class Yellow implements TrafficLight {
+  isRed() { return false; }
+  isYellow() { return true; }
+  isGreen() { return false; }
+  updateCar() { return car.drive(); }
+}
+
+class Green implements TrafficLight {
+  isRed() { return false; }
+  isYellow() { return false; }
+  isGreen() { return true; }
+  updateCar() { return car.drive(); }
+}
+
+const car = new Car();
+const CYCLE = [new Red(), new Green(), new Yellow()];
+const updateCar = (car: TrafficLight) => car.updateCar();
```

Comment:
```suggestion
const updateCar = (car: TrafficLight) => car.updateCar();
```
매개 변수는 클래스와 함수가 아닌 인터페이스로 매개변수를 사용하고 있습니다. 왜 인터페이스를 사용 했을까요? 이 부분을 찾아보고 공부를 해보면 어떨까요?

Reply:
인터페이스를 매개변수로 선언하면 해당 인터페이스를 기반으로 구현한 
여러 객체를 매개변수에 할당해줄 수 있기 때문? 이라고 생각합니다.

Reply:
말씀해주신 부분도 맞지만 더 중요한 개념은 **다형성**이 있습니다. 이 부분은 매우 중요한 개념이라 반드시 공부를 해보시면 좋겠습니다!!

Reply:
아 terminology 의 중요성을 다시 한번 깨닫고 갑니다.. 감사합니다